DynamicActiveQuery.php

```diff
@@ -48,13 +87,18 @@ public function prepare($builder)
             );
         }
 
-        if (empty($this->select) || $this->select === '*') {
-            $this->select = array_diff(array_keys($modelClass::getTableSchema()->columns), [$this->dynamicColumn]);
+        if (empty($this->select)) {
+            $this->select[] = '*';
+        }
+
+        if (is_array($this->select) && in_array('*', $this->select)) {
             $this->db = $modelClass::getDb();
             $this->select[$this->dynamicColumn] =
                 'COLUMN_JSON(' . $this->db->quoteColumnName($this->dynamicColumn) . ')';
         }
 
+        $this->preProcessDynamicAttributes();
```

Comment:
the pre and post processing (in the private methods below) seems heavy. are there no alternatives? i don't mind changing the token characters.